addon/array/-utils.js

```diff
@@ -18,7 +18,9 @@ function convertArray(array) {
   }
 
   if (Array.isArray(array)) {
-    return emberA(array);
+    // we use .slice() here so that plain arrays are
+    // not internally mutated to Ember arrays
+    return emberA(array.slice());
   }
 
   if (isEmberArray(array)) {
```

Comment:
Looks to me that this section is never used. `Array.isArray` is true for ember arrays too.

Reply:
only for Ember Arrays that are backed by native arrays

Reply:
I just commented the whole block out and all the tests still pass.

Reply:
hmm, I'm very surprised by that. I wrote the comment below because when I only returned `array` the `concat` test started failing 🤔

Reply:
nvm. My rebuilds aren't very trustworthy apparently.